### PR TITLE
Make compliance changes to ManagedStorageAccount KeyVault cmdlets

### PR DESCRIPTION
**Pull request by maddieclayton**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/770/

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageAccount.cs [hunk 1]
```diff
@@ -18,16 +18,19 @@
...
 using KeyVaultProperties = Microsoft.Azure.Commands.KeyVault.Properties;
 using Microsoft.Azure.Commands.KeyVault.Models.ManagedStorageAccounts;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
 
 namespace Microsoft.Azure.Commands.KeyVault
 {
-    [Cmdlet( VerbsCommon.Get, CmdletNoun.AzureKeyVaultManagedStorageAccount)]
-    [OutputType( typeof( List<PSKeyVaultManagedStorageAccount> ), typeof( PSKeyVaultManagedStorageAccount ), typeof(List<PSDeletedKeyVaultManagedStorageAccountIdentityItem>), typeof(PSDeletedKeyVaultManagedStorageAccount) )]
+    [Cmdlet( VerbsCommon.Get, CmdletNoun.AzureKeyVaultManagedStorageAccount, DefaultParameterSetName = ByAccountNameParameterSet)]
+    [OutputType( typeof( List<PSKeyVaultManagedStorageAccountIdentityItem> ), typeof( PSKeyVaultManagedStorageAccount ), typeof(List<PSDeletedKeyVaultManagedStorageAccountIdentityItem>), typeof(PSDeletedKeyVaultManagedStorageAccount) )]
```
Comment (markcowl) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageAccount.cs]: we should try to have a single output type, if there is a common ancestor type, and it should be a single type, not an enumeration

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageSasDefinition.cs [hunk 2]
```diff
@@ -21,13 +21,14 @@
 
 namespace Microsoft.Azure.Commands.KeyVault
 {
-    [Cmdlet( VerbsCommon.Get, CmdletNoun.AzureKeyVaultManagedStorageSasDefinition)]
+    [Cmdlet( VerbsCommon.Get, CmdletNoun.AzureKeyVaultManagedStorageSasDefinition, DefaultParameterSetName = ByDefinitionNameParameterSet)]
```
Comment (markcowl) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageSasDefinition.cs]: same comment on OutputType

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageAccount.cs (hunk 1 above)
Comment (dragav) [reply src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageAccount.cs]: Re-reading my late night comment, I realize that is the very definition of a base type; I meant that deleted full-entity types are not direct descendants of full-entity types.

Comment (maddieclayton) [reply src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/GetAzureKeyVaultManagedStorageAccount.cs]: @dragav It looks like all of the return types are descendants of KVEntityIdentityItem - changing the "OutputType" to be one concrete type will allow users to tab through all the properties of KVEntityIdentityItem when running a cmdlet like "select -Properties", while having multiple output types gives them the option to tab through all properties of all output types, which in this case could be confusing when they are retrieving a regular managedstorageaccount and get the properties for a deleted managedstorageaccount while tabbing.  Note: they can still retreive these properties by typing them out, but they will not be available through tabbing.

**Comment by dragav** [comment]
If that means that, upon invoking GetDeletedMSA -Name <msaName>, the user can only tab through the properties of the EntityIdentityItem type, then I find that equally confusing/unhelpful.

**Comment by maddieclayton** [comment]
@dragav If you'd rather it return all possible properties, I will do that - I'll update it in the next commit.

**Pull request merged**